apps/cowswap-frontend/src/modules/usdAmount/apis/getBffUsdPrice.ts

```diff
@@ -0,0 +1,64 @@
+import { BFF_BASE_URL } from '@cowprotocol/common-const'
+import { FractionUtils } from '@cowprotocol/common-utils'
+import { Fraction, Token } from '@uniswap/sdk-core'
+
+import ms from 'ms.macro'
+
+import { fetchWithRateLimit } from 'common/utils/fetch'
+
+import { UnknownCurrencyError } from './errors'
+
+type BffUsdPriceResponse = {
+  price: number
+}
+type BffUsdErrorResponse = { message: string }
+type BffResponse = BffUsdPriceResponse | BffUsdErrorResponse
+
+const BASE_URL = BFF_BASE_URL
+
+const fetchRateLimited = fetchWithRateLimit({
+  // Allow 5 requests per second
+  rateLimit: {
+    tokensPerInterval: 5,
+    interval: 'second',
+  },
+  // 2 retry attempts with 100ms delay
+  backoff: {
+    maxDelay: ms`0.1s`,
+    numOfAttempts: 2,
+  },
+})
+
+export async function getBffUsdPrice(currency: Token): Promise<Fraction | null> {
+  const url = `${BASE_URL}/${currency.chainId}/tokens/${currency.address}/usdPrice`
+
+  return fetchRateLimited(url)
+    .then((res) => res.json())
+    .catch((error) => {
+      if (error.message === 'Price not found') {
+        // TODO: will a 404 response be caught here?
```

Comment:
you can probably test by changing the endpoint name

Reply:
It seems fragile to rely on the message, no?

Reply:
Better to rely on response code? (404)?